### ISSUE TITLE
I am not able to user onDestroy() lifecycle hook in my flutter application? Can you tell me what is the workaround to this?

### ISSUE THREAD

**Issue by aditya234** [flutter/flutter#21982]

**Comment by dineshba** [comment]
Could you please give some more details?

**Comment by aditya234** [comment]
Actually, I wanted my application to perform some tasks everytime it is closed, even if its closed forcefully. 
To do that, I tried to tap into onDestroy() lifecycle method, by creating a plugin. Since no other option was known to me and I certainly didn't find it that easy on Official docs.
Please help me, I am new to flutter.

**Comment by zoechi** [comment]
Can you please provide the information I asked for ;-)

**Comment by aditya234** [comment]
Adityas-MacBook-Air-2:~ adityas$ flutter doctor -v
[✓] Flutter (Channel beta, v0.7.3, on Mac OS X 10.13.6 17G65, locale en-IN)
    • Flutter version 0.7.3 at /Users/adityas/developments/flutter
    • Framework revision 3b309bda07 (4 weeks ago), 2018-08-28 12:39:24 -0700
    • Engine revision af42b6dc95
    • Dart version 2.1.0-dev.1.0.flutter-ccb16f7282

[!] Android toolchain - develop for Android devices (Android SDK 28.0.2)
    • Android SDK at /Users/adityas/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[!] iOS toolchain - develop for iOS devices (Xcode 9.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.4.1, Build version 9F2000
    ✗ libimobiledevice and ideviceinstaller are not installed. To install, run:
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    ✗ ios-deploy not installed. To install:
        brew install ios-deploy
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 27.1.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] IntelliJ IDEA Community Edition (version 2018.2.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 28.0.4
    • Dart plugin version 182.4129.13

[✓] VS Code (version 1.27.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.18.0

[!] Connected devices
    ! No devices available

! Doctor found issues in 3 categories.

**Comment by aditya234** [comment]
Yeah, It worked while the thread was suspended. But this method does not do the job when the app is terminated purposefully.
Anyways, thanks for the help.
I think there is no way to do it in Flutter.

**Comment by zoechi** [comment]
> I think there is no way to do it in Flutter.

I rather think there is no way to do it in iOS.

**Comment by aditya234** [comment]
Yes, I also read in a few articles saying that no such hooks exist in IOS, while some say that such an arrangement can be made.
But, I think this type of hooks must be provided.

**Comment by Navneet-chaurasia** [comment]
this is happening to me also and I think xyx.luan.audioplayers is causing this.

**Comment by tbm98** [comment]
currently, I finding in [AppLifecycleState](https://api.flutter.dev/flutter/dart-ui/AppLifecycleState-class.html) but not found destroy state
